builder/frameworks/arduino.py

```
@@ -24,254 +24,989 @@
+
+    print(f"*** Framework includes found: {framework_count}/"
+          f"{len(includes)} ***")
+
+    # Show path distribution analysis
+    analysis = analyze_path_distribution(includes)
+    print(f"*** Path Analysis: Avg={analysis.get('average_length', 0):.1f}, "
+          f"Max={analysis.get('max_length', 0)}, "
+          f"Framework Avg={analysis.get('framework_avg_length', 0):.1f} ***")
+
+
+def apply_include_shortening(env, node, includes, total_length):
```

Pylint (3.3.7)

[refactor] Too many local variables (20/15)

(R0914)